controller/aks-cluster-config-handler.go

```diff
@@ -0,0 +1,686 @@
+		if err != nil {
+			logrus.Errorf("Error removing resource group [%s] message: %v", config.Spec.ResourceGroup, err)
+			return config, err
+		}
+	}
+
+	logrus.Infof("Deployment was removed successfully")
+	return config, nil
+}
+
+// recordError writes the error return by onChange to the failureMessage field on status. If there is no error, then
+// empty string will be written to status
+func (h *Handler) recordError(onChange func(key string, config *aksv1.AKSClusterConfig) (*aksv1.AKSClusterConfig, error)) func(key string, config *aksv1.AKSClusterConfig) (*aksv1.AKSClusterConfig, error) {
+	return func(key string, config *aksv1.AKSClusterConfig) (*aksv1.AKSClusterConfig, error) {
+		var err error
+		var message string
+		config, err = onChange(key, config)
+		if config == nil {
+			// EKS config is likely deleting
+			return config, err
+		}
+		if err != nil {
+			if !strings.Contains(err.Error(), "currently has update") {
+				// the update is valid in that the controller should retry but there is
+				// no actionable resolution as far as a user is concerned. An update
+				// that has either been initiated by eks-operator or another source is
+				// already in progress. It is possible an update is not being immediately
+				// reflected in upstream cluster state. The config object will reenter the
+				// controller and then the controller will wait for the update to finish.
+				message = err.Error()
+			}
+		}
+
+		if config.Status.FailureMessage == message {
+			return config, err
+		}
+
+		config = config.DeepCopy()
+		if message != "" {
+			if config.Status.Phase == aksConfigActivePhase {
+				// can assume an update is failing
+				config.Status.Phase = aksConfigUpdatingPhase
+			}
+		}
+		config.Status.FailureMessage = message
+
+		var recordErr error
+		config, recordErr = h.aksCC.UpdateStatus(config)
+		if recordErr != nil {
+			logrus.Errorf("Error recording akscc [%s] failure message: %s", config.Name, recordErr.Error())
+		}
+		return config, err
+	}
+}
+
+func (h *Handler) CreateDeployment(ctx context.Context, config *aksv1.AKSClusterConfig) (*aksv1.AKSClusterConfig, error) {
+
+	logrus.Infof("Creating deployment for config [%s]", config.Spec.DisplayName)
+
+	azureAuthorizer, _ := newClientAuthorizer(config.Spec)
+
+	resourceGroupsClient, _ := newResourceGroupsClient(azureAuthorizer, config.Spec)
+	groupExists, _ := checkResourceGroup(ctx, resourceGroupsClient, config.Spec)
+
+	logrus.Infof("Checking if resource group [%s] exists", config.Spec.ResourceGroup)
+
+	if !groupExists {
+		logrus.Infof("Creating resource group [%s] for config [%s]", config.Spec.ResourceGroup, config.Spec.DisplayName)
+		status, err := createResourceGroup(ctx, resourceGroupsClient, config.Spec)
+		if !status || err != nil {
+			logrus.Errorf("Error creating resource group [%s] with message %v", config.Spec.ResourceGroup, err)
+		}
+		logrus.Infof("Resource grup [%s] created successfully", config.Spec.ResourceGroup)
```

Comment:
Is the resource group created successfully if an error occurs? Wondering if we need an `else` here instead, or to re-write this a different way... or I could just be misunderstanding.